internal/guest/oom/oom.go

```diff
@@ -0,0 +1,80 @@
+// Package oom providers wrappers and support to leverage the OOM watcher functionality
+// in github.com/containerd/containerd/pkg/oom
+package oom
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/Microsoft/hcsshim/internal/guest/prot"
+	log "github.com/Microsoft/hcsshim/internal/log"
+	eventstypes "github.com/containerd/containerd/api/events"
+	"github.com/containerd/containerd/events"
+	"github.com/containerd/containerd/runtime"
+	"github.com/containerd/containerd/runtime/v2/shim"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+)
+
+var emptyActivityID = (uuid.UUID{}).String()
```

Comment:
I wouldn't rely on the uuid package here, we were looking to get rid of it. It's just an all zero guid, so you could just hardcode it here for now.